main/integration-tests/__test__/api-tests/data-sources/register-ds-bucket.test.js

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const { runSetup } = require('../../../support/setup');
+const errorCode = require('../../../support/utils/error-code');
+
+describe('Register data source bucket scenarios', () => {
+  let setup;
+  let adminSession;
+  let accountId;
+
+  beforeAll(async () => {
+    setup = await runSetup();
+    adminSession = await setup.defaultAdminSession();
+    accountId = setup.gen.accountId();
+    await adminSession.resources.dataSources.accounts.create({ id: accountId });
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
```

Comment:
Could you confirm cleanup is working as expected? (I was seeing some leftover resources on the DDB post-test last time I ran these tests)

Reply:
Yes, needed to do that as part of writing the cleanup logic for the data source. I see all the necessary entries in the dsAccounts and RoleAllocations tables being correctly cleaned up.